orchagent/vnetorch.cpp

```diff
@@ -0,0 +1,200 @@
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to remove virtual router name: %s, rv:%d",
+                            vnet_name.c_str(), status);
+        }
+    }
+
+    SWSS_LOG_INFO("VNET '%s' deleted ", vnet_name.c_str());
+}
+
+/*
+ * VNet Orch class definitions
+ */
+
+template <class T>
+std::unique_ptr<T> VNetOrch::createObject(const string& vnet_name, set<string>& plist,
+                                          vector<sai_attribute_t>& attrs)
+{
+    std::unique_ptr<T> vnet_obj(new T(vnet_name, plist, attrs));
+    return vnet_obj;
+}
+
+bool VNetOrch::addOperation(const Request& request)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+    vector<sai_attribute_t> attrs;
+    set<string> peer_list = {};
+    bool peer = false, create = false;
+
+    for (const auto& name: request.getAttrFieldNames())
+    {
+        if (name == "src_mac")
+        {
+            const auto& mac = request.getAttrMacAddress("src_mac");
+            attr.id = SAI_VIRTUAL_ROUTER_ATTR_SRC_MAC_ADDRESS;
+            memcpy(attr.value.mac, mac.getMac(), sizeof(sai_mac_t));
+        }
+        else if (name == "peer_list")
+        {
+            peer_list  = request.getAttrSet("peer_list");
+            peer = true;
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Logic error: Unknown attribute: %s", name.c_str());
+            continue;
+        }
+        attrs.push_back(attr);
+    }
+
+    const std::string& vnet_name = request.getKeyString(0);
+    SWSS_LOG_INFO("VNET '%s' add request", vnet_name.c_str());
+
+    try
+    {
+        VNetObject_T obj;
+        auto it = vnet_table_.find(vnet_name);
+        if (isVnetExecVrf())
+        {
+            if (it == std::end(vnet_table_))
+            {
+                obj = createObject<VNetVrfObject>(vnet_name, peer_list, attrs);
+                create = true;
+            }
+            SWSS_LOG_NOTICE("VNET '%s' was added ", vnet_name.c_str());
+        }
+        else
+        {
+            // BRIDGE Handling
+        }
+
+        if (create)
+        {
+            vnet_table_[vnet_name] = std::move(obj);
+        }
+        else if (peer)
+        {
+            it->second->setPeerList(peer_list);
+        }
+        else if (!attrs.empty())
+        {
+            if(!it->second->updateObj(attrs))
+            {
+                return true;
+            }
+        }
+
+    }
+    catch(std::runtime_error& _)
+    {
+        SWSS_LOG_ERROR("VNet add operation error for %s: error %s ", vnet_name.c_str(), _.what());
+        return false;
+    }
+
+    SWSS_LOG_INFO("VNET '%s' added/updated ", vnet_name.c_str());
+    return true;
+}
+
+bool VNetOrch::delOperation(const Request& request)
+{
+    SWSS_LOG_ENTER();
+
+    const std::string& vnet_name = request.getKeyString(0);
+
+    if (vnet_table_.find(vnet_name) == std::end(vnet_table_))
+    {
+        SWSS_LOG_ERROR("VNET '%s' doesn't exist", vnet_name.c_str());
```

Comment:
>SWSS_LOG_ERROR [](start = 8, length = 14)

not sure this error, it should be warning since the program still can run without problem.